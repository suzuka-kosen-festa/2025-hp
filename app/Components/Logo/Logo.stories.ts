import type { Meta, StoryObj } from "@storybook/react";

import Logo from "./Logo";

type T = typeof Logo;

const meta: Meta = {
  title: "Logo",
  component: Logo,
  parameters: { layout: "centered" },
  tags: ["autodocs"],
  argTypes: {
    length: {
      control: { type: "number" },
      description: "Logo length of a side",
    },
    simpleLogo: {
      control: { type: "boolean" },
      description: "Choose between full logo or simple logo",
    },
    withBlackBackground: {
      control: { type: "boolean" },
      description: "Add black background to the logo",
    },
    withStars: {
      control: { type: "boolean" },
      description: "Add stars background layer",
    },
  },
} satisfies Meta<T>;

type Story = StoryObj<T>;

export const Default: Story = {};

export const SimpleLogo: Story = {
  args: { simpleLogo: true },
};

export const WithBlackBackground: Story = {
  args: { withBlackBackground: true },
};

export const SimpleLogoWithBlackBackground: Story = {
  args: { simpleLogo: true, withBlackBackground: true },
};

export const WithStars: Story = {
  args: { withStars: true },
};

export const SimpleLogoWithStars: Story = {
  args: { simpleLogo: true, withStars: true },
};

export const WithBlackBackgroundAndStars: Story = {
  args: { withBlackBackground: true, withStars: true },
};

export const SimpleLogoWithBlackBackgroundAndStars: Story = {
  args: {
    simpleLogo: true,
    withBlackBackground: true,
    withStars: true,
  },
};

export const SmallSize: Story = {
  args: { length: 300 },
};

export const LargeSize: Story = {
  args: { length: 800 },
};

export default meta;
