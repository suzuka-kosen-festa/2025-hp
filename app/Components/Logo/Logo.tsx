import type { ReactNode } from "react";
import { Box } from "@mui/material";

interface LogoProps {
  length?: string | number
  simpleLogo?: boolean
  withBlackBackground?: boolean
  withStars?: boolean
}

function Logo({
  length = 500,
  simpleLogo = false,
  withBlackBackground = false,
  withStars = false,
}: LogoProps): ReactNode {
  const logoSrc = simpleLogo ? "/simple-logo.svg" : "/logo.svg";

  // レスポンシブなサイズを計算
  const responsiveLength = typeof length === "number"
    ? { xs: Math.min(length * 0.6, 300), sm: Math.min(length * 0.8, 400), md: length }
    : length;

  return (
    <Box
      sx={{
        position: "relative",
        display: "inline-block",
        width: responsiveLength,
        height: responsiveLength,
        backgroundColor: withBlackBackground ? "#000000" : "transparent",
        borderRadius: "8px",
      }}
    >
      <img
        src={logoSrc}
        alt="Logo"
        style={{
          position: "relative",
          width: "100%",
          height: "100%",
          objectFit: "contain",
          zIndex: 2,
        }}
      />
      {withStars && (
        <img
          src="/stars.svg"
          alt="Stars background"
          style={{
            position: "absolute",
            top: 0,
            left: 0,
            width: "100%",
            height: "100%",
            objectFit: "cover",
            zIndex: 1,
          }}
        />
      )}
    </Box>
  );
}

export default Logo;
