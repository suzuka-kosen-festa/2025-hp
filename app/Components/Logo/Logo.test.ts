// @vitest-environment jsdom

import { composeStories } from "@storybook/react";
import { expect, it } from "vitest";

import * as stories from "./Logo.stories";

const composedStories = composeStories(stories);
Object.entries(composedStories).forEach(([storyName, Story]) => {
  it(`logo snapshot: ${storyName}`, async () => {
    await Story.run();
    expect(document.body.firstChild).toMatchSnapshot();
  });
});
