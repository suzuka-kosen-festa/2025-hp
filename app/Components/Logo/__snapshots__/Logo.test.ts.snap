// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`logo snapshot: Default 1`] = `
<div>
  <div
    class="MuiBox-root css-rsa0i0"
  >
    <img
      alt="Logo"
      src="/logo.svg"
      style="position: relative; width: 100%; height: 100%; object-fit: contain; z-index: 2;"
    />
  </div>
</div>
`;

exports[`logo snapshot: LargeSize 1`] = `
<div>
  <div
    class="MuiBox-root css-9m3z7"
  >
    <img
      alt="Logo"
      src="/logo.svg"
      style="position: relative; width: 100%; height: 100%; object-fit: contain; z-index: 2;"
    />
  </div>
</div>
`;

exports[`logo snapshot: SimpleLogo 1`] = `
<div>
  <div
    class="MuiBox-root css-rsa0i0"
  >
    <img
      alt="Logo"
      src="/simple-logo.svg"
      style="position: relative; width: 100%; height: 100%; object-fit: contain; z-index: 2;"
    />
  </div>
</div>
`;

exports[`logo snapshot: SimpleLogoWithBlackBackground 1`] = `
<div>
  <div
    class="MuiBox-root css-1o0i47h"
  >
    <img
      alt="Logo"
      src="/simple-logo.svg"
      style="position: relative; width: 100%; height: 100%; object-fit: contain; z-index: 2;"
    />
  </div>
</div>
`;

exports[`logo snapshot: SimpleLogoWithBlackBackgroundAndStars 1`] = `
<div>
  <div
    class="MuiBox-root css-1o0i47h"
  >
    <img
      alt="Logo"
      src="/simple-logo.svg"
      style="position: relative; width: 100%; height: 100%; object-fit: contain; z-index: 2;"
    />
    <img
      alt="Stars background"
      src="/stars.svg"
      style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; z-index: 1;"
    />
  </div>
</div>
`;

exports[`logo snapshot: SimpleLogoWithStars 1`] = `
<div>
  <div
    class="MuiBox-root css-rsa0i0"
  >
    <img
      alt="Logo"
      src="/simple-logo.svg"
      style="position: relative; width: 100%; height: 100%; object-fit: contain; z-index: 2;"
    />
    <img
      alt="Stars background"
      src="/stars.svg"
      style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; z-index: 1;"
    />
  </div>
</div>
`;

exports[`logo snapshot: SmallSize 1`] = `
<div>
  <div
    class="MuiBox-root css-1lenzg4"
  >
    <img
      alt="Logo"
      src="/logo.svg"
      style="position: relative; width: 100%; height: 100%; object-fit: contain; z-index: 2;"
    />
  </div>
</div>
`;

exports[`logo snapshot: WithBlackBackground 1`] = `
<div>
  <div
    class="MuiBox-root css-1o0i47h"
  >
    <img
      alt="Logo"
      src="/logo.svg"
      style="position: relative; width: 100%; height: 100%; object-fit: contain; z-index: 2;"
    />
  </div>
</div>
`;

exports[`logo snapshot: WithBlackBackgroundAndStars 1`] = `
<div>
  <div
    class="MuiBox-root css-1o0i47h"
  >
    <img
      alt="Logo"
      src="/logo.svg"
      style="position: relative; width: 100%; height: 100%; object-fit: contain; z-index: 2;"
    />
    <img
      alt="Stars background"
      src="/stars.svg"
      style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; z-index: 1;"
    />
  </div>
</div>
`;

exports[`logo snapshot: WithStars 1`] = `
<div>
  <div
    class="MuiBox-root css-rsa0i0"
  >
    <img
      alt="Logo"
      src="/logo.svg"
      style="position: relative; width: 100%; height: 100%; object-fit: contain; z-index: 2;"
    />
    <img
      alt="Stars background"
      src="/stars.svg"
      style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; z-index: 1;"
    />
  </div>
</div>
`;
